service/src/main/java/bio/terra/user/app/controller/ProfileApiController.java

```diff
@@ -19,32 +21,44 @@ public class ProfileApiController implements ProfileApi {
   private final HttpServletRequest servletRequest;
   private final ProfileService profileService;
   private final ProfileSamUserFactory userFactory;
+  private final SamService samService;
 
   @Autowired
   public ProfileApiController(
       HttpServletRequest servletRequest,
       ProfileService profileService,
-      ProfileSamUserFactory userFactory) {
+      ProfileSamUserFactory userFactory,
+      SamService samService) {
     this.servletRequest = servletRequest;
     this.profileService = profileService;
     this.userFactory = userFactory;
+    this.samService = samService;
   }
 
   @Override
-  public ResponseEntity<AnyObject> setUserProfile(AnyObject body, String path) {
-    profileService.setProperty(getUser(), parsePath(path), body.getValue());
+  public ResponseEntity<AnyObject> setUserProfile(AnyObject body, String path, String userEmail) {
+    profileService.setProperty(targetIdOrSelf(userEmail), parsePath(path), body.getValue());
 
-    return getUserProfile(path);
+    return getUserProfile(path, userEmail);
   }
 
   @Override
-  public ResponseEntity<AnyObject> getUserProfile(String path) {
+  public ResponseEntity<AnyObject> getUserProfile(String path, String userEmail) {
     AnyObject apiObj =
-        new AnyObject().value(profileService.getProperty(getUser(), parsePath(path)));
+        new AnyObject()
+            .value(profileService.getProperty(targetIdOrSelf(userEmail), parsePath(path)));
 
     return new ResponseEntity<>(apiObj, HttpStatus.OK);
   }
 
+  private String targetIdOrSelf(String userEmail) {
+    return StringUtils.isEmpty(userEmail)
+        ? getUser().getSubjectId()
+        : SamRethrow.onInterrupted(
+            () -> samService.adminEmailToId(getUser().getBearerToken(), userEmail),
+            "Admin user Id from email");
```

Comment:
rephrase it to something like If user is admin, get the targeted user id from email.

It's easy to misread this as getting the admin's user id.